File: DAY_13/Devraj/Daily_Temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n=temperatures.size();
        vector<int>res(n,0);
        stack<pair<int,int>>s;
        s.push({temperatures[n-1],n-1});
        for(int i=n-2;i>=0;i--)
        {    while(!s.empty() && s.top().first<=temperatures[i])
              s.pop();
             if(s.empty())
                 res[i]=0;
             else
                 res[i]=s.top().second -i;
            s.push({temperatures[i],i});
                
        }
        return res;
        
    }
};

File: DAY_13/Ravi/Daily Temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n=temperatures.length;
        int[] ans = new int[n];
        Stack<Integer> st = new Stack<>();
        for(int i=n-1;i>=0;i--) {
            while(!st.isEmpty() && temperatures[i]>=temperatures[st.peek()])
                st.pop();
            ans[i]=st.isEmpty()?0:st.peek()-i;
            st.push(i);
        }
        return ans;
    }
}
